### DISCUSSION TITLE
Course Completion

### DISCUSSION THREAD

**Discussion by dodeyeikoi** [Cyfrin/foundry-full-course-cu#735] [Q&A] [answered]
Hello sir Patrick and all the expert developers, I have a question. Upon completion of this course will I be able to do any of the following:

1. Design and develop smart contracts: Create self-executing contracts that automate business logic and processes.

2. Code optimization: Write efficient and gas-efficient code to reduce transaction costs on the blockchain.

3. Security: Implement security best practices to prevent vulnerabilities and attacks like reentrancy, overflow, and more.

4. Integration: Integrate smart contracts with front-end applications, databases, and other external systems.

5. Testing: Write unit and integration tests to ensure the functionality and security of your smart contracts.

6. Deployment: Deploy smart contracts to the appropriate blockchain network, such as Ethereum or other platforms.

7. Maintenance: Monitor and update smart contracts as needed based on changing requirements or vulnerabilities

**Comment by Niferu** [comment, marked as answer]
Hello @dodeyeikoi 

YES!

To all points expect below:

4. This course has not a lot of integration with front-end. To get more knowledge on that and learn frontend integration you should take this course: [link](https://github.com/smartcontractkit/full-blockchain-solidity-course-js)
7. Smart contracts are immutable, you can just upgrade them into newer versions, but those will be other contracts already. You will learn everything about that in lesson 13 from this course

**Reply by dodeyeikoi** [reply]
Thank you! @Neftyr I look forward to completing the course

**Reply by Niferu** [reply]
@dodeyeikoi 

Glad I could help!

Please mark this as answer, so other with similar question can see this ;)

ps. Java Script version of this course is amazing fullfilling front-end needs
